wasi/wasi.go

```diff
@@ -503,6 +520,22 @@ func (a *api) fd_close(ctx *wasm.HostFunctionCallContext, fd uint32) (err Errno)
 	return ErrnoSuccess
 }
 
+func (a *api) RandomGet(ctx *wasm.HostFunctionCallContext, buf uint32, buf_len uint32) (errno Errno) {
+	if !ctx.Memory.ValidateAddrRange(buf, uint64(buf_len)) {
+		return ErrnoInval
+	}
+
+	random_bytes := make([]byte, buf_len)
+	_, err := ctx.Randomizer.Read(random_bytes)
```

Comment:
```suggestion
	_, err := a.randSource.Read(random_bytes)
```

Reply:
once you have `randSource` in wasi.api struct, then the code look like this ^^ and you don't need to expose the WASI concepts into wasm space